### ISSUE TITLE
decltype(auto) is misinterpreted in function definition

### ISSUE THREAD

**Issue by rempke** [doxygen/doxygen#7309]
When defining a forward declared function with return type `decltype(auto)`, doxygen 1.8.15 with default configuration is confused by the return type:
```c++
decltype(auto) Function();

decltype(auto) Function()
{ }
```

```
Decltype.hpp:3: warning: documented symbol `decltype' was not declared or defined.
```
[Example](https://github.com/doxygen/doxygen/files/3712106/decltype.zip)

**Comment by bbczeuz** [comment]
Over a year old: https://sourceforge.net/p/doxygen/mailman/doxygen-users/thread/20180419061941.0A828347F059%40macserver.private/

Proposed workaround: Use PREDEFINED for decltype

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.9.3.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
